doc/api/process.md

```diff
@@ -1996,6 +1996,136 @@ Will generate an object similar to:
   tz: '2016b' }
 ```
 
+## Access Control
+
+> Stability: 1 - Experimental
+
+This feature is experimental and needs to be enabled by passing
+the `--experimental-access-control` flag, or the
+`--disable=restrictionA,restrictionB,...` flag to Node.js.
+
+The `--disable` flag takes a comma-separated list of identifiers as described
+below. For example, it can be used as `--disable=fsRead,fsWrite`.
+
+### process.accessControl.apply(restrictions)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `restrictions` {Object} A set of access restrictions that will be applied
+  to the current Node.js instance. The format should be the same as the one
+  returned by [`process.accessControl.getCurrent()`][]. Omitted keys will
+  default to `true`, i.e. to retaining the relevant permissions.
+
+*Warning*: This does not provide a full isolation mechanism. Child processes,
+native addons, and other communication channels may be used to circumvent
+these measures, if they have been made available before the corresponding
+restrictions have been put into place.
+
+This API is recent and may not be complete.
+Please report bugs at https://github.com/nodejs/node/issues.
+
+Operations started before this call are not undone or stopped.
+Features that were previously disabled through this call cannot be re-enabled.
+Child processes do not inherit these restrictions, whereas [`Worker`][]s do.
```

Comment:
This should be clarified. For instance, if I use `fs.createWriteStream()` now to open a file for writing, then set the access control policy, will I be able to use the write stream or no?

Another question: if the Node.js process stdout/stderr are redirected to a pipe and fsWrite is denied, what happens when the process attempts to write out?

Reply:
@jasnell This is a very good point.

> For instance, if I use `fs.createWriteStream()` now to open a file for writing, then set the access control policy, will I be able to use the write stream or no?

You will not, because of the way fs streams are implemented – as individual `fs.read` and `fs.write` operations in JS.

However, existing network sockets can still be used, because we implement them in a very abstract way in C++ that is not specific to the kind of resource (e.g. `net` & `tty` all use the same code). We can align behaviours if we think that that makes sense, though.

I’ve pushed docs & test changes to account for these scenarios.

> Another question: if the Node.js process stdout/stderr are redirected to a pipe and fsWrite is denied, what happens when the process attempts to write out?

Assuming you meant “redirected to a file”: An exception occurs. `console.log` and friends silence those, but using raw `process.stdout.write()` will throw.